pkg/resources/pool.go

```diff
@@ -0,0 +1,119 @@
+// Copyright 2018 Intel Corp. All Rights Reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package resources
+
+import (
+	"fmt"
+
+	"github.com/golang/glog"
+	"github.com/intel/sriov-network-device-plugin/pkg/types"
+	"github.com/intel/sriov-network-device-plugin/pkg/utils"
+	pluginapi "k8s.io/kubernetes/pkg/kubelet/apis/deviceplugin/v1beta1"
+)
+
+const (
+	sysBusPci        = "/sys/bus/pci/devices"
+	totalVfFile      = "sriov_totalvfs"
+	configuredVfFile = "sriov_numvfs"
+)
+
+type resourcePool struct {
+	config      *types.ResourceConfig
+	devices     map[string]*pluginapi.Device
+	bindScript  string
```

Comment:
Initially, I was thinking of do the driver binding unbinding using the dpdk-devbind.py script if that was needed. But this may add additional complexities and issues. It's better to leave the driver binding part outside of the scope of device plugin. This way device plugin can stay as simple as possible. Left the binding code here should we need in the future. Or else can be removed.